Rollbar/RollbarLogger.cs

```diff
@@ -0,0 +1,282 @@
+        }
+
+        public ILogger Debug(string msg, IDictionary<string, object> custom = null)
+        {
+            return this.Log(ErrorLevel.Debug, msg, custom);
+        }
+
+        public ILogger Debug(ITraceable traceableObj, IDictionary<string, object> custom = null)
+        {
+            return this.Debug(traceableObj.TraceAsString(), custom);
+        }
+
+        public ILogger Debug(object obj, IDictionary<string, object> custom = null)
+        {
+            return this.Debug(obj.ToString(), custom);
+        }
+
+        public ILogger Error(string msg, IDictionary<string, object> custom = null)
+        {
+            return this.Log(ErrorLevel.Error, msg, custom);
+        }
+
+        public ILogger Error(System.Exception error, IDictionary<string, object> custom = null)
+        {
+            this.Report(error, ErrorLevel.Error, custom);
+
+            return this;
+        }
+
+        public ILogger Info(string msg, IDictionary<string, object> custom = null)
+        {
+            return this.Log(ErrorLevel.Info, msg, custom);
+        }
+
+        public ILogger Warning(string msg, IDictionary<string, object> custom = null)
+        {
+            return this.Log(ErrorLevel.Warning, msg, custom);
+        }
+
+        #endregion ILogger
+
+        private Guid? Report(System.Exception e, ErrorLevel? level = ErrorLevel.Error, IDictionary<string, object> custom = null)
+        {
+            return SendBody(new Body(e), level, custom);
+        }
+
+        private Guid? Report(string message, ErrorLevel? level = ErrorLevel.Error, IDictionary<string, object> custom = null)
+        {
+            return SendBody(new Body(new Message(message)), level, custom);
+        }
+
+        private Guid? SendBody(Body body, ErrorLevel? level, IDictionary<string, object> custom)
+        {
+            if (string.IsNullOrWhiteSpace(this._config.AccessToken) 
+                || this._config.Enabled == false
+                )
+            {
+                return null;
+            }
+
+            var guid = Guid.NewGuid();
+
+            var data = new Data(this._config.Environment, body)
+            {
+                Custom = custom,
+                Level = level ?? this._config.LogLevel
+            };
+
+            var payload = new Payload(this._config.AccessToken, data);
+            payload.Data.GuidUuid = guid;
+            payload.Data.Person = this._config.Person;
+
+            if (this._config.Server != null)
+            {
+                payload.Data.Server = this._config.Server;
+            }
+
+            if (this._config.CheckIgnore != null 
+                && this._config.CheckIgnore.Invoke(payload)
+                )
+            {
+                return null;
+            }
+
+            this._config.Transform?.Invoke(payload);
+
+            this._config.Truncate?.Invoke(payload);
+
+            this._payloadQueue.Enqueue(payload);
+
+            //var client = new RollbarClient(_config);
```

Comment:
This has all been moved into `RollbarQueueController` if I am following along correctly?

Reply:
correct. all the fun now happens within the controller, that watches all the queues because it is also responsible for keeping accumulative rate of reports below access token's limit (across all the queue targeting the same token).  
removed the commented out code.